starfish/image/_filter/bandpass.py

```diff
@@ -112,6 +112,7 @@ def run(
             in_place=in_place,
             n_processes=n_processes,
         )
+        result.update_log(self)
```

Comment:
I was trying to think of a way to do this. But I think the logging needs to happen in the run method so that it get's run with both the api and cli. And even if I added it to the base class I'd still have to put super.run() everywhere so I wasn't sure if that was better.

Reply:
Mild idea: use a decorator.
Wild idea: can we use a metaclass to detect that someone has subclassed, and provided a run method, and inject in the necessary logic?

Reply:
Note that a decorator does not significantly improve the situation in terms of LOC or having to remember to update things.  It might be a bit cleaner though.